packages/design-system/src/scalars/components/url-field/url-field.tsx

```diff
@@ -0,0 +1,94 @@
+import React, { useId } from "react";
+import {
+  Input,
+  FormGroup,
+  withFieldValidation,
+  FormLabel,
+  FormDescription,
+  FormMessageList,
+} from "../fragments";
+import { ErrorHandling, FieldCommonProps } from "../types";
+import { cn } from "@/scalars/lib";
+
+interface UrlFieldProps
+  extends FieldCommonProps<string>,
+    ErrorHandling,
+    Omit<
+      React.InputHTMLAttributes<HTMLInputElement>,
+      "pattern" | "value" | "defaultValue" | "name"
+    > {
+  allowedProtocols?: string[];
+}
+
+const UrlFieldRaw: React.FC<UrlFieldProps> = ({
+  label,
+  description,
+  warnings,
+  errors,
+  allowedProtocols = ["http", "https"],
+  ...props
+}) => {
+  const idGenerated = useId();
+  const id = props.id ?? idGenerated;
+
+  const showProtocol = allowedProtocols.length > 1;
+  const protocol = `${allowedProtocols[0] ?? "https"}://`;
+
+  return (
+    <FormGroup>
+      <FormLabel
+        htmlFor={id}
+        required={props.required}
+        disabled={props.disabled}
+        hasError={!!errors?.length}
+      >
+        {label}
+      </FormLabel>
+      <div className={"flex items-center"}>
+        {showProtocol && (
+          <Input
+            value={protocol}
+            readOnly
+            // tailwind does not support dynamic classes
+            style={{ width: `${(protocol.length + 1) * 8}px` }}
+            className={`w-auto rounded-r-none border-r-0 focus:z-10`}
+          />
+        )}
+        <Input
+          id={id}
+          type="url"
+          role="textbox"
```

Comment:
Perhaps we don’t need `role="textbox"` here? An <input type="url"> is already accessible without it.

Reply:
I remove it, thanks!